doc/fundamentals/incrementality.md

```diff
@@ -0,0 +1,86 @@
+# Incrementality
+
+Pachyderm performs computations in an incremental fashion, that is, rather
+than computing a result all at once, it computes it in small pieces and
+then stitches those pieces together. This allows Pachyderm to compute
+things much more efficiently by reusing things that it's already computed
+rather than computing them again. There are two forms of incrementality
+that Pachyderm supports, this doc will cover them both and explain how to
+leverage them to speed up your workload.
+
```

Comment:
Maybe list both types here and link to the sections.  This seems to work well in other docs.